### PR TITLE
Assignment 8

### PR DESCRIPTION
**Pull request by Fervidusletum**

### CODE REVIEW

**Comment by MeguminXIII** [comment]
I will review

**Review by MeguminXIII — COMMENTED** [review]
Assignment
Update the GroupRepository with functionality to generate gift assignments. ✔
A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
A user is not allowed to be both the Giver and Recipient of the assignment. ❌
need to implement this. just gotta check if the user is already a giver/recipient and throw an error if it is.

Update the GroupsController with an endpoint for generating the assignments. ✔
Update the Groups/Edit page to contain the UI elements for the following
A means to generate assignments. ✔
A means to display any errors that occurs while generating assignments. ✔
Display of the assignments for each of the users. ✔

Write appropriate unit tests for changes made to the GroupRepository ❌
Write appropriate functional tests for the change made to the GroupsControllers❌
need to implement

Extra Credit
There is currently no way to for a user to add gifts. On the edit user page add functionality to be able to add a gift to the user. ❌
Allow for navigating to a user from the Group edit page to make it easy to see the gifts requested by a user. ❌

looks good, need to implement tests and put in the functionality for making it so a user cannot be both a giver and a recipient. Otherwise it looks good.

File: SecretSanta/src/SecretSanta.Business/GroupRepository.cs [hunk 1]
```diff
@@ -45,5 +52,30 @@ public void Save(Group item)
...
+        {
+            Group? group = GetItem(id);
+            if (group?.Users is null) return AssignmentResult.Error("Group not found");
+            if (group.Users.Count < 3) return AssignmentResult.Error($"Group {group?.Name} must have at least three users");
+
+            group.Assignments.Clear();
+
+            
+            List<User> shuffled = ShuffleList(group.Users);
+            shuffled.Add(shuffled.First()); // add first user to end too, so we dont need extra circular link logic
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Business/GroupRepository.cs:66]: Clever implementation

**Review by Keboo — COMMENTED** [review]
# Assignment
- Update the `GroupRepository` with functionality to generate gift assignments. ✔
  - A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
  - A user is not allowed to be both the Giver and Recipient of the assignment. ✔
- Update the GroupsController with an endpoint for generating the assignments. ✔
- Update the Groups/Edit page to contain the UI elements for the following
  - A means to generate assignments. ✔
  - A means to display any errors that occurs while generating assignments. ✔
  - Display of the assignments for each of the users. ✔
- Write appropriate unit tests for changes made to the `GroupRepository` ✔
- Write appropriate functional tests for the change made to the `GroupsController` ✔

**Pull request closed without merging**